### DISCUSSION TITLE
Virtualization of the DOM/data for million rows

### DISCUSSION THREAD

**Discussion by Sahil333** [dream-num/univer#1109] [Q&A | 常见问题]
Does/Will this lib support virtualization when the number of rows are too many, say a million? If one wants to implement it, how to go about it?

**Comment by wzhudev** [comment]
We have virtual scroll working on canvas just like virtual scroll for DOM. Only rows in the viewport would be rendered.

**Reply by Sahil333** [reply]
What about lazy loading of the data only into the worksheet, not talking about the rendering? First 10k, then next 20k and at some point removing data from the top to keep memory usage of the tab under a certain limit.
